### PR TITLE
FSE Plugin: new patterns, fix minor typo

### PR DESCRIPTION
**Pull request by ianstewart**
This PR adds eight new patterns to the FSE Plugin and corrects one minor typo.

cc @johnHackworth @iamtakashi @obenland

### CODE REVIEW

File: apps/full-site-editing/full-site-editing-plugin/block-patterns/patterns/recent-posts.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+<!-- /wp:jetpack/layout-grid-column --></div>
+<!-- /wp:jetpack/layout-grid -->
+';
+
+return array(
+	'__file'  => 'wp_block',
+	'title'   => esc_html__( 'Recent Posts', 'full-site-editing' ),
+	'content' => sprintf(
+		$markup
+	),
```
Comment (obenland) [inline apps/full-site-editing/full-site-editing-plugin/block-patterns/patterns/recent-posts.php:21]: If there's nothing to replace, feel free to pass `$markup` directly
```suggestion
	'content' => $markup,
```

**Comment by ianstewart** [comment]
@obenland with that update from your review (thanks, btw!) I'm just going to make one more test on WP.com and then merge this.

**Comment by obenland** [comment]
Tested on dotcom and they all seem to work as expected.

**Comment by obenland** [comment]
This is what it looks like for me on an Atomic site:
image

The call-to-action-02 pattern is still broken and one of the contact ones is, too:
> Your site doesn’t include support for the "jetpack/map" block. 

Could be because the site is not connected to dotcom or because of the plan maybe?

**Comment by ianstewart** [comment]
The map being missing is strange. It _should_ be there and available. I'm still confused about what's happening with the broken Call to Action block.

**Comment by iamtakashi** [comment]
> what's happening with the broken Call to Action block.

I made the same button from scratch in both Simple site and Atomic site, and the markup that I've got is _slightly_ different, and that's probably why it slowing the error.

**Simple Site**
```
<!-- wp:buttons {"align":"center"} -->
<div class="wp-block-buttons aligncenter"><!-- wp:button {"customBackgroundColor":"#a5150f","customTextColor":"#ffffff"} -->
<div class="wp-block-button"><a class="wp-block-button__link has-text-color has-background" style="background-color:#a5150f;color:#ffffff">Order now</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons -->
```

**Atomic**
```
<!-- wp:buttons {"align":"center"} -->
<div class="wp-block-buttons aligncenter"><!-- wp:button {"customBackgroundColor":"#a5150f","customTextColor":"#ffffff"} -->
<div class="wp-block-button" style="color:#ffffff"><a class="wp-block-button__link has-text-color has-background" style="background-color:#a5150f;color:#ffffff">Order now</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons -->
```
But I don't know why they are different. Do Atomic and Simple use different versions of Gutenberg?

**Comment by iamtakashi** [comment]
I copied the markup of the pattern with a map from the pattern master site to an Atomic site, and I didn't see any issue 🤔

**Comment by ianstewart** [comment]
> I copied the markup of the pattern with a map from the pattern master site (Simple site, premium plan) to an Atomic site (Business plan), and I didn't see any issue 🤔

I had the same experience with an eCommerce plan just now.

**Comment by andrewserong** [comment]
I think the `jetpack/map` issue is particular to testing on a fresh ephemeral site where you're setting up Jetpack for the first time. When I tested on an ephemeral site, before setting up Jetpack, I got the same error message about the block being unsupported as in the screenshots above.

After activating Jetpack, it then displays the box for entering a Mapbox token as in [the support docs](https://jetpack.com/support/jetpack-blocks/map-block/):

image

If I then add in a Mapbox token, the `jetpack/map` block starts working correctly (and in the block inserter):

image

In these screenshots I'm testing with Gutenberg v8.0.0, too, and I noticed that the text is displaying as black within this block. If I go to re-create the block manually, this doesn't happen, so I'm wondering if the markup might change a little between the versions? Would it be worth re-creating the blocks again in v8.0.0?

As for dealing with block unsupported messages, would it make sense to have our custom patterns use a check to see if blocks in `jetpack` or `coblocks` are available before adding them into the list? This might be an edge case, but just thinking about Atomic users who might deactivate Coblocks for some reason, and then they'd see broken patterns.

**Comment by andrewserong** [comment]
I've done some more digging into the issue with the call to action 2 broken pattern on Atomic (https://github.com/Automattic/wp-calypso/issues/41380), and it looks like the markup that ends up getting on a simple site after inserting the pattern is different to what's in `call-to-action-02.php`. I'm wondering if this is because on simple sites we're running the [code to fix invalid blocks](https://github.com/automattic/wp-calypso/blob/master/apps/wpcom-block-editor/src/wpcom/features/fix-block-invalidation-errors.js#L47), which we're not running on Atomic sites. So the markup that ends up being in the simple site has the correct inline style transformations, which would explain the difference @iamtakashi [commented on](https://github.com/Automattic/wp-calypso/pull/41636#issuecomment-621507957).

The solution is to update the markup, and I guess a simple way to do that would be to add the pattern on a simple site, then go to the visual editor and copy the markup from there. I've pasted the markup that works for me on the tracking issue https://github.com/Automattic/wp-calypso/issues/41380

I'm thinking that a similar solution might be needed for the black paragraph text in the contact 3 pattern.

That's about all the time I have for investigating this issue for today, but I'll add this PR to the Create Focus maintenance board, so hopefully the next set of Create team gardeners might have some time to chip in, too.

**Comment by ianstewart** [comment]
> The solution is to update the markup, and I guess a simple way to do that would be to add the pattern on a simple site, then go to the visual editor and copy the markup from there.

This was how the block pattern template was created.

**Comment by andrewserong** [comment]
> This was how the block pattern template was created.

Great point. I think was way off in my hunch from the other day when it comes to the [broken button block](https://github.com/Automattic/wp-calypso/issues/41380) in `call-to-action-02`. I can see that the pattern renders correctly without error if I disable CoBlocks in my Atomic site, which points to a conflict in the full build of CoBlocks we're using there. I added a bit more info for where to try looking deeper into this [in the issue](https://github.com/Automattic/wp-calypso/issues/41380#issuecomment-623271292).

**Comment by ianstewart** [comment]
@andrewserong the call to action pattern looks like it'll be corrected with https://github.com/Automattic/wp-calypso/pull/41870 I think that just leaves the Jetpack Map Block to correct. I'll "rebuild" the map block with 8.0 and update the PR. 🤞:)

**Comment by ianstewart** [comment]
> I did notice that if I don't have any posts published in my site then the latest posts patterns show a "Sorry, no posts were found" error, but this feels perfectly fine to me:

I believe there's an issue for this elsewhere. Yes, it's weird. :)

**Pull request merged**